KamerConnect.DataAccess.Postgres/Repositories/ChatRepository.cs

```diff
@@ -72,28 +75,220 @@ INSERT INTO chatmessages (
+                        {
+                            AddPersonToChat(chatId, personId);
+                        }
+                    }
+                    catch
+                    {
+                        transaction.Rollback();
+                        throw;
+                    }
+                }
+
+            }
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine($"Error while creating chats: {e.Message}");
+            throw;
+        }
+    }
+
+    private List<Guid> GetChatIdsFromPerson(Guid personId)
+    {
+        using (var connection = new NpgsqlConnection(_connectionString))
+        {
+            connection.Open();
+
+            using (var command = new NpgsqlCommand("""
+                                                   SELECT DISTINCT id
+                                                   FROM person_chat pc
+                                                   LEFT JOIN chat c ON pc.chat_id = c.id
+                                                   WHERE (pc.person_id = @personId::uuid)
+                                                   """,
+                       connection))
+            {
+                command.Parameters.AddWithValue("@personId", personId);
+
+                using (var reader = command.ExecuteReader())
+                {
+                    List<Guid> chatIds = new List<Guid>();
+
+                    while (reader.Read())
+                    {
+                        chatIds.Add(reader.GetGuid(0));
+                    }
+
+                    return chatIds;
+                }
+            }
+        }
+    }
+
+    private Guid? GetMatchIdFromChat(Guid chatId)
+    {
+        using (var connection = new NpgsqlConnection(_connectionString))
+        {
+            connection.Open();
+
+            using (var command = new NpgsqlCommand("""
+                                                   SELECT match_id
+                                                   FROM chat c
+                                                   WHERE (c.id = @chatId::uuid)
+                                                   """,
+                       connection))
+            {
+                command.Parameters.AddWithValue("@chatId", chatId);
+
+                using (var reader = command.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        return reader.GetGuid(0);
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    public List<Chat> GetChatsFromPersonId(Guid personId)
+    {
+        try
+        {
+            using (var connection = new NpgsqlConnection(_connectionString))
+            {
+                connection.Open();
+
+                using (var transaction = connection.BeginTransaction())
+                {
+                    try
+                    {
+                        List<Guid> chatIds = GetChatIdsFromPerson(personId);
```

Comment:
GetChatIdsFromPerson kan dat niet gewoon zijn GetChatsFromPerson? Dan hoef je alleen maar het matchId erbij te geven en krijg en dan hoef je ook geen losse get matches te doen. Ook als er in de toekomst nog velden bij een chat komen hoef je dan niet nog een losse functie te maken.

Reply:
maar op die manier hoe zou ik dan de personen er bij koppelen voor in het Chat object? Ik moet ook nog alle personen erbij vinden

Reply:
Personen kunnen in 1e instantie een lege list zijn die daarna vult of overschrijft